central/testutils/testexport.go

```diff
@@ -0,0 +1,348 @@
+		),
+	)
+	h.pool = pgtest.ForT(tb)
+	deploymentStore, err := deploymentDataStore.GetTestPostgresDataStore(tb, h.pool)
+	if err != nil {
+		return err
+	}
+	h.Deployments = deploymentStore
+	h.Images = imageDataStore.GetTestPostgresDataStore(tb, h.pool)
+	return nil
+}
+
+// TearDownTest cleans up the ExportServicePostgresTestHelper resources after testing.
+func (h *ExportServicePostgresTestHelper) TearDownTest(tb testing.TB) {
+	h.pool.Teardown(tb)
+	h.pool.Close()
+}
+
+// getAuthInterceptor returns a GRPC Steam Interceptor that overrides the server context to make sure all calls are authenticated.
+func getAuthInterceptor(ctx context.Context) func(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	return func(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+		return handler(srv, &grpcMiddleware.WrappedServerStream{
+			ServerStream:   ss,
+			WrappedContext: ctx,
+		})
+	}
+}
+
+// CreateGRPCStreamingService creates a streaming server, registers the target
+// services there, and returns a connection to the streaming server along with
+// a function to close the connection.
+func (h *ExportServicePostgresTestHelper) CreateGRPCStreamingService(
+	_ testing.TB,
+	registerServices func(registrar grpc.ServiceRegistrar),
+) (*grpc.ClientConn, func(), error) {
+	bufferSize := 1024 * 1024
+	listener := bufconn.Listen(bufferSize)
+
+	server := grpc.NewServer(grpc.StreamInterceptor(getAuthInterceptor(h.Ctx)))
+	registerServices(server)
+
+	go func() {
+		utils.IgnoreError(func() error { return server.Serve(listener) })
+	}()
+
+	conn, err := grpc.DialContext(h.Ctx, "",
+		grpc.WithContextDialer(
+			func(ctx context.Context, _ string) (net.Conn, error) {
+				return listener.DialContext(ctx)
+			},
+		),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	closeFunc := func() {
+		utils.IgnoreError(listener.Close)
+		server.Stop()
+	}
+	return conn, closeFunc, nil
+}
+
+func getImageSetPath() (string, error) {
+	// Go up the directory tree from the current working directory
+	// to location where the subtree to the image data file matches.
+	cwd, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+	pathElems := strings.Split(cwd, string(os.PathSeparator))
+	for i := len(pathElems); i >= 0; i-- {
+		basePath := strings.Join(pathElems[:i], string(os.PathSeparator))
+		imageDataPathElems := []string{basePath, "central", "testutils", "testdata", "imgdata.json.gz"}
+		imageDataPath := strings.Join(imageDataPathElems, string(os.PathSeparator))
+		_, err := os.Stat(imageDataPath)
+		if err == nil {
+			return imageDataPath, nil
+		}
+		if os.IsNotExist(err) {
+			continue
+		}
+		return "", err
+	}
+	return "", nil
+}
+
+// GetBaseImageSet returns a set of realistic images for testing purposes.
+func GetBaseImageSet() ([]*storage.Image, error) {
```

Comment:
What is the "BaseImageSet"? Let's just call it `GetTestImages` instead? You're not returning a `set` but just an array.

Reply:
I re-used a list of image names from some scanner tests, had them scanned, exported, and used the exported data to create test data.